### DISCUSSION TITLE
Update sidebar to correctly identify progress on jumps and scrolling

### DISCUSSION THREAD

**Discussion by Luna-Klatzer** [pradyunsg/furo#207] [Ideas] [answered]
# Summary
While writing some docs and generating some smaller sections I noticed that the sidebar was not moving as I expected, so that every item is iterated through, but it actually jumps around and partially skips smaller sections entirely through the screen touching other elements as well.

*Example: (Add Para-C as alias) is entirely skipped even when slowly moving*

https://user-images.githubusercontent.com/61756091/127367971-e83a5593-5e3b-4e75-b44f-6267adf828e0.mp4

This is kind of irritating while using and in bigger forms, where loads of smaller items/headers are used this can lead to huge jumps, where progress does not appear clean and is almost random.

## Idea
- As a fix for clicking: When clicking identifying the exact header and highlighting that exact item until scrolling was activated where it can be ignored. Aka. adding some form of overwriting that lasts until scrolling was activated or the user jumped somewhere else to keep the marked item highlighted.
- As a fix for scrolling: Possibly adding a movement animation, which uses every entry to scroll through, so that entries are not skipped. This could be possibly ignored on jumps, where you do not scroll.

**Comment by pradyunsg** [comment]
This theme uses https://github.com/cferdinandi/gumshoe for the sidebar scrollspy. That project, however, seems pretty dead. If you have any "scrollspy" libraries, that are vanilla JS or jQuery based that have better/configurable behaviours like the ones you're looking for, I'm very interested! :)

**Reply by Luna-Klatzer** [reply]
I sadly don't have any, but I just noticed that the scrolling seemed a bit off when using it. It seems though it should be not very hard to implement due to having to track only the current window position according to the highest header, and implementing special handling if the user clicked on a header. Sadly I do not have the time to implement something like that, but maybe I'll look into it if I find something or can quickly write something that could fit that.

**Comment by pradyunsg** [comment, marked as answer]
Suggestions and PRs for adopting a jQuery / vanilla JS sidebar scrollspy library are welcome!
